### PR TITLE
V2 Part A

### PR DESCRIPTION
**Pull request by Venomalia**
Part A contains everything except the SubRegion support.

grafik

v2 core changes some of the structure, so some general functions are added like that regions are no longer added manually to the OwnedTexture. however, nothing is changed in the ui.

Region commands, Region commands allow you to update and delete regions via the toolbar.

make delete buttons uniform, the "X" button in HostKeyList and EmulatetKeyList is replaced by the Delete icon

### CODE REVIEW

File: DolphinDynamicInputTexture/Data/DynamicInputPropertys.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿namespace DolphinDynamicInputTexture.Propertys
```
Comment (iwubcode) [inline DolphinDynamicInputTexture/Data/DynamicInputPropertys.cs:1]: Minor but can we change this to `Properties` (and update the filename as well).

File: DolphinDynamicInputTexture/Data/InputRegionRect.cs [hunk 2]
```diff
@@ -125,56 +96,76 @@ public double Width
...
+            return false;
         }
 
-        [JsonIgnore]
-        public double ScaledY
+        public bool IntersectsWith(IRectRegion other)
         {
-            get => Y * ScaleFactor;
-            set => Y = value / ScaleFactor;
+            if (X > other.RightX || Y > other.BottomY || RightX < other.X || BottomY < other.Y)
```
Comment (iwubcode) [inline DolphinDynamicInputTexture/Data/InputRegionRect.cs:146]: The ifs in these functions can go probably away:

```
public bool IntersectsWith(IRectRegion other) => !(X > other.RightX || Y > other.BottomY || RightX < other.X || BottomY < other.Y);
```

File: ViewModels/TagsViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            {
+                _tags = value;
+                TagsSuggestions.SetTargetList(Tags);
+                OnPropertyChanged(nameof(Tags));
+            }
+        }
+        private UICollection<Tag> _tags;
+
+        /// <summary>
+        /// possible key Suggestions for the Selected emulated device.
```
Comment (iwubcode) [inline ViewModels/TagsViewModel.cs:30]: Summary probably needs to be updated here :)

**Comment by iwubcode** [comment]
This now needs a rebase :)

File: ViewModels/TagsViewModel.cs (hunk 3 above)
Comment (Venomalia) [reply ViewModels/TagsViewModel.cs]: have changed everything as suggested. :)

**Pull request merged**